tests/Unit/Domain/FunctionsOfTime/Test_QuaternionFunctionOfTime.cpp

```diff
@@ -31,7 +31,70 @@ void test_serialization_versioning() {
 
   TestHelpers::serialization::test_versioning<QuatFoT>(
       "Domain/FunctionsOfTime/QuaternionFunctionOfTime.serializations",
-      "version 4", func);
+      "version 5", func);
+}
+
+void test_out_of_order_update() {
+  using QuatFoT = domain::FunctionsOfTime::QuaternionFunctionOfTime<2>;
+  register_classes_with_charm<QuatFoT>();
+  const DataVector quat_dv{1.0, 0.0, 0.0, 0.0};
+  const DataVector zero_dv{3, 0.0};
+  const std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime> func(
+      std::make_unique<QuatFoT>(
+          0.0, std::array<DataVector, 1>{quat_dv},
+          std::array<DataVector, 3>{zero_dv, zero_dv, zero_dv}, 0.5));
+
```

Comment:
can you add another update `func->update(2.5, zero_dv, 3.0)` here? This way we ensure that the `update_backlog` can store multiple entries that are given out of order. This should work because the `std::map` is guaranteed to be ordered but I had to think about it a bit  so its good to have a quick test.